### PR TITLE
fix: settings with unsavedchanges dialog

### PR DESCRIPTION
**Pull request by Kiebert**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

Replaced the standard Prompt with the UnSavedChangedDialog component.

## Git Issues

[#2061](https://github.com/ONEARMY/community-platform/issues/2061)

### CODE REVIEW

**Comment by davehakkens** [comment]
Thanks @Kiebert!
You want to give a try to see @MIfoodie to see if this fixes your problem?

File: src/pages/Settings/SettingsPage.tsx [hunk 1]
```diff
@@ -186,8 +186,8 @@ export class UserSettings extends React.Component<IProps, IState> {
           const heading = user.profileType ? 'Edit profile' : 'Create profile'
           return (
             <Flex mx={-2} bg={'inherit'} sx={{ flexWrap: 'wrap' }}>
-              <Prompt
-                when={!this.injected.userStore.updateStatus.Complete}
+              <UnsavedChangesDialog
+                uploadComplete={this.injected.userStore.updateStatus.Complete}
                 message={
                   'You are leaving this page without saving. Do you want to continue ?'
                 }
```
Comment (thisislawatts) [inline src/pages/Settings/SettingsPage.tsx:193]: **question:** @Kiebert did you look at introducing a failing test that captures this bug, which then passes as a result of the changes you have made here?

File: src/common/Form/UnsavedChangesDialog.tsx [hunk 2]
```diff
@@ -39,7 +41,7 @@ export const UnsavedChangesDialog = memo((props: IProps) => {
         render={() => null}
       />
       {/* Render a blocking screen prompt to prevent user leaving page via react navigation */}
-      <Prompt when={shouldPromptUnsavedChanges} message={CONFIRM_DIALOG_MSG} />
+      <Prompt when={shouldPromptUnsavedChanges} message={message} />
```
Comment (thisislawatts) [inline src/common/Form/UnsavedChangesDialog.tsx:44]: **praise:** Great to see this component extended for use in another place in the Platform. Thanks @Kiebert ✨

File: src/pages/Settings/SettingsPage.tsx (hunk 1 above)
Comment (Kiebert) [reply src/pages/Settings/SettingsPage.tsx:193]: Hi @thisislawatts. I think I can create a test like that.

**Comment by davehakkens** [comment]
Did a little test here, but not sure what changed with the current live site. 
Could you explain or share some screenshots of the change @Kiebert?

**Comment by Kiebert** [comment]
@davehakkens The test.app website expired. I will first write the test upcoming weekend. 
After that I will check the newly generated web.app.

File: src/pages/Settings/SettingsPage.tsx (hunk 1 above)
Comment (Kiebert) [reply src/pages/Settings/SettingsPage.tsx:193]: Hi @thisislawatts. I added two tests and when the confirmation is triggered is produces a log in the loglist of Cypress.
Is there another way to check is the confirmation window is active or not?

**Comment by Kiebert** [comment]
> Did a little test here, but not sure what changed with the current live site. Could you explain or share some screenshots of the change @Kiebert?

I added the tests. 
So after this change the confirmation window should on only appear when you want to navigate to a different page but you have changed/edited data in the form.

When no edit has to be done in the form and you want to navigate to a different page the confirmation window should not appear.

File: src/pages/Settings/SettingsPage.tsx (hunk 1 above)
Comment (thisislawatts) [reply src/pages/Settings/SettingsPage.tsx:193]: Let me take a look @Kiebert and see what I can find.

Comment (thisislawatts) [reply src/pages/Settings/SettingsPage.tsx:193]: Did you see the tests you added fail @Kiebert? I lifted just the 2 new test cases onto the `master` branch and ran them. They both passed, which suggests they are not capturing the buggy behaviour.

I've experimented with Cypress + browser alert modals before, but it looks like theres a path forward here:
https://stackoverflow.com/a/73086525

**Comment by davehakkens** [comment]
> So after this change the confirmation window should on only appear when you want to navigate to a different page but you have changed/edited data in the form.

Thanks for clarifying. Here I currently get no confirmation window at all. When I fill in half my profile and click another page like `how-to` or `event` it just goes there and removes the data.
Is this different when you test @Kiebert ?

**Comment by Kiebert** [comment]
> > So after this change the confirmation window should on only appear when you want to navigate to a different page but you have changed/edited data in the form.
> 
> Thanks for clarifying. Here I currently get no confirmation window at all. When I fill in half my profile and click another page like `how-to` or `event` it just goes there and removes the data. Is this different when you test @Kiebert ?

I am getting this confirmation window when I am changing for example my 'Display Name':
afbeelding

@davehakkens Which browser do you use? And what happens when you clear your browser cache?

**Comment by davehakkens** [comment]
got it. works now. For some reason the local testing didn't work well but the preview link does.
Nice little fix!

File: src/pages/Settings/SettingsPage.tsx (hunk 1 above)
Comment (thisislawatts) [reply src/pages/Settings/SettingsPage.tsx:193]: Nothing from my side, it looks good. Thanks for the follow up, we need one of the current maintainers to review. @davehakkens

**Review by davehakkens — APPROVED** [review]

**Pull request merged**